### PR TITLE
Newmodule clippy

### LINKED ISSUE #3369: new module: CLIPPY
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by CharlotteAnne**
## PR checklist

Closes #3369  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/clippy/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
+process CLIPPY {
+    tag "$meta.id"
+    label "process_high"
```
Comment (chris-cheshire) [inline modules/nf-core/clippy/main.nf:3]: Does it def need high?

File: tests/config/test_data.config [hunk 2]
```diff
@@ -184,6 +184,7 @@ params {
                 genome_21_sdf                                  = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome_sdf.tar.gz"
                 genome_21_fasta                                = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta"
                 genome_21_fasta_fai                            = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai"
+                genome_21_gtf                                  = "${params.test_data_base}/data/genomics/homo_sapiens/genome/chr21/sequence/chr21_gencode.gtf"
```
Comment (chris-cheshire) [inline tests/config/test_data.config:187]: I think you need to name this explicitly gencode, also is there anything else different about this gtf file?

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

File: modules/nf-core/clippy/main.nf (hunk 1 above)
Comment (CharlotteAnne) [reply modules/nf-core/clippy/main.nf:3]: yes we had issues on large files if set lower

File: tests/config/test_data.config (hunk 2 above)
Comment (CharlotteAnne) [reply tests/config/test_data.config]: nah its just gencode chr21 - renamed now!

File: tests/modules/nf-core/clippy/test.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- name: clippy test_clippy
```
[the remark this answers is not in the document]
Comment (CharlotteAnne) [reply tests/modules/nf-core/clippy/test.yml:1]: good call! added a test for this!

File: tests/config/pytest_modules.yml [hunk 4]
```diff
@@ -3644,6 +3648,10 @@ ultra/pipeline:
   - modules/nf-core/ultra/pipeline/**
   - tests/modules/nf-core/ultra/pipeline/**
 
+umicollapse:
```
[the remark this answers is not in the document]
Comment (CharlotteAnne) [reply tests/config/pytest_modules.yml]: removed!

**Comment by CharlotteAnne** [comment]
addressed all comments!

File: modules/nf-core/clippy/main.nf [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    conda "bioconda::clippy=1.5.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/clippy:1.5.0--pyhdfd78af_0' :
+        'quay.io/biocontainers/clippy:1.5.0--pyhdfd78af_0' }"
```
Comment (charles-plessy) [inline modules/nf-core/clippy/main.nf:8]: Please remove `quay.io/` as nf-core tools 2.8 do not need it anymore.

**Review by chris-cheshire — APPROVED** [review]

**Pull request merged**